### PR TITLE
Look for rel=me on the root of a domain when user logs in with a path.

### PR DESCRIPTION
**Pull request by mblaney**

### CODE REVIEW

File: models.py [hunk 1]
```diff
@@ -559,11 +560,26 @@ def _urls_and_domains(self, auth_entity, user_url):
     for i, url in enumerate(candidates):
       final, domain, ok = util.get_webmention_target(url, resolve=i < MAX_AUTHOR_URLS)
       if ok:
-        if util.schemeless(final.lower()).startswith(util.schemeless(url.lower())):
+        current_url = ''
+        final = final.lower()
+        # Ignore redirects when fetching the root url, they are dealt with next.
+        if util.schemeless(final) == util.schemeless(url.lower()):
+          # If final has a path segment check if root has a matching rel=me.
+          match = re.match(r'^(https?://[^/]+)/.+', final)
```
Comment (snarfed) [inline models.py]: heh. i started to ask for `urlparse` instead, but then realized you also need to extract just the root URL. doable with `urlparse` too, but annoying. this lgtm. :P

File: models.py [hunk 2]
```diff
@@ -559,11 +560,26 @@ def _urls_and_domains(self, auth_entity, user_url):
     for i, url in enumerate(candidates):
       final, domain, ok = util.get_webmention_target(url, resolve=i < MAX_AUTHOR_URLS)
       if ok:
-        if util.schemeless(final.lower()).startswith(util.schemeless(url.lower())):
+        current_url = ''
+        final = final.lower()
+        # Ignore redirects when fetching the root url, they are dealt with next.
```
Comment (snarfed) [inline models.py]: hmm. we probably still want do this new rel-me check even after a lateral redirect, though, right? e.g. if we start with`/foo` and get redirected to `/bar`?

File: models.py [hunk 3]
```diff
@@ -559,11 +560,26 @@ def _urls_and_domains(self, auth_entity, user_url):
...
+        if util.schemeless(final) == util.schemeless(url.lower()):
```
Comment (snarfed) [inline models.py]: we should also gate on `i < MAX_AUTHOR_URLS`, like with `resolve=` above, to limit the number of different root URLs we fetch. feel free to put it in a temporary variable if you want, e.g. `resolve`.

File: models.py (hunk 2 above)
Comment (mblaney) [reply models.py]: yes, thanks for pointing it out. I realised there's another case too, going from a url with a path to a deeper path... then I realised all redirects are actually just a pre-check, so the whole thing became simpler! :-)

File: models.py (hunk 3 above)
Comment (mblaney) [reply models.py]: yes should've picked that up too. I just added it to `if match:`, do you want that to be `if match and resolve:`?

**Comment by snarfed** [comment]
woo, thank you! looks great, merging.

**Pull request merged**